contracts/LightClient.sol

```diff
@@ -121,4 +129,54 @@ contract LightClient is ILightClient, Ownable {
     function proposedValidators() external view override returns (address[] memory, uint256[] memory) {
         return staking.proposedValidators();
     }
+
+    function getEpochIdx(uint256 height) public view override returns (uint256) {
+        require(isInHeightRange(height), "out of height range");
+        uint256 _epochPeriod = epochPeriod;
+        return (height + _epochPeriod - 1) / epochPeriod;
+    }
+
+    function isInHeightRange(uint256 height) public view override returns (bool) {
+        (uint256 min, uint256 max) = heightRange();
+        if (height >= min && height <= max) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function minEpochIdx() public view override returns (uint256) {
+        if (curEpochIdx >= TOTAL_EPOCH) {
+            return curEpochIdx - TOTAL_EPOCH + 1;
+        }
+        return 0;
+    }
+
+    function heightRange() public view override returns (uint256 min, uint256 max) {
+        min = _minHeight();
+        max = _maxHeight();
+        // when the curEpochHeight is 10000 and TOTAL_EPOCH = 2,
```

Comment:
I feel the name `curEpochHeight` is pretty misleading - it is the highest height of the last epoch (instead of the current Epoch).

Reply:
I guess this one is not yet resolved?